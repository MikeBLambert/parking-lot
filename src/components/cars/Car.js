import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Car = ({ cars }) => {

  const carItems = cars.map(car => {
    return (
      <tr key={car.plate}>
        <td>{car.make}</td>
        <td>{car.model}</td>
        <td>{car.plate}</td>
      </tr>
    );
    // <li key={car.plate}>Make: {car.make} Model: {car.model} Plate: {car.plate}</li>;
  });

  return (
    <Fragment>
      {carItems}
    </Fragment>
  );
};

Car.propTypes = {
  cars: PropTypes.array.isRequired
};

export default Car;
