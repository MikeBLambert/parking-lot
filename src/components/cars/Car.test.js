import React from 'react';
import { shallow } from 'enzyme';
import Car from './Car';

describe('Car component', () => {

  const cars = [{ make: 'Toyota', model: 'Corolla', plate: 'ABC123' }];
  
  it('matches snapshot', () => {
    const wrapper = shallow(<Car cars={cars}/>);
    expect(wrapper).toMatchSnapshot();
  });


});
