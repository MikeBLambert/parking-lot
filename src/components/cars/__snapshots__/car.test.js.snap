// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car component matches snapshot 1`] = `
<Fragment>
  <tr
    key="ABC123"
  >
    <td>
      Toyota
    </td>
    <td>
      Corolla
    </td>
    <td>
      ABC123
    </td>
  </tr>
</Fragment>
`;
